external-resolver/src/main/scala/org/genivi/sota/resolver/DeviceDataMigrator.scala

```diff
@@ -0,0 +1,44 @@
+package org.genivi.sota.resolver
+
+import java.time.Instant
+
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.data.{PackageId, Uuid}
+import org.genivi.sota.resolver.db.ForeignPackages.InstalledForeignPackage
+import org.genivi.sota.resolver.db.{DeviceRepository, ForeignPackages}
+import org.slf4j.LoggerFactory
+import slick.driver.MySQLDriver.api._
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+object DeviceDataMigrator {
+
+  private lazy val logger = LoggerFactory.getLogger(this.getClass)
+
+  def apply(deviceRegistry: DeviceRegistry)(implicit db: Database, ec: ExecutionContext): Unit = {
+    db.run(packages()).onComplete {
+      case Success(rows) =>
+        val packagesPerDevice = mutable.Map[Uuid, Seq[PackageId]]()
+        rows.foreach { elem =>
+          packagesPerDevice(elem.device) = packagesPerDevice.getOrElse(elem.device, List()) :+ elem.packageId
+        }
+        deviceRegistry.setInstalledPackagesForDevices(packagesPerDevice.toSeq).onComplete {
+          case Failure(ex) =>
+            logger.error(s"Failed to write ${packagesPerDevice.toString()} to resolver: ${ex.toString}")
+          case Success(_) =>
+        }
+      case ex =>
+        logger.error(s"failed for reason ${ex.toString}")
+    }
+  }
+
+  def packages()(implicit db: Database, ec: ExecutionContext): DBIO[Seq[InstalledForeignPackage]] =
+    for {
+      nonForeignPkgRows <- DeviceRepository.listInstalledPackages
+      //use sets here to remove duplicates
+      nonForeignPkgs = nonForeignPkgRows.map{r => InstalledForeignPackage(r._2, r._3, Instant.now())}.toSet
```

Comment:
Need to use a constant time can't use `Instant.now()` since it will be different for each element in the set (and therefore not remove duplicates of device, packageId).